intg-denonavr/denon_remote.py

```diff
@@ -0,0 +1,1226 @@
+        """Initialize the class."""
+        self._device: avr.DenonDevice = receiver
+        self._denon_media_player: DenonMediaPlayer = denon_media_player
+        entity_id = create_entity_id(receiver.id, EntityTypes.REMOTE)
+        features = [Features.SEND_CMD, Features.ON_OFF, Features.TOGGLE]
+        super().__init__(
+            entity_id,
+            f"{device.name} Remote",
+            features,
+            attributes={
+                Attributes.STATE: receiver.state,
+            },
+            simple_commands=simplecommand.get_simple_commands(device),
+            button_mapping=REMOTE_BUTTONS_MAPPING,
+            ui_pages=DenonRemote._get_remote_ui_pages(),
+        )
+
+    async def command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        """
+        Remote entity command handler.
+
+        Called by the integration-API if a command is sent to a configured remote entity.
+
+        :param cmd_id: command
+        :param params: optional command parameters
+        :return: status code of the command request
+        """
+        return await self._handle_command(cmd_id, params)
+
+    @staticmethod
+    def state_from_avr(avr_state: avr.States) -> ucapi.remote.States:
+        """
+        Convert AVR state to UC API remote state.
+
+        :param avr_state: Denon/Marantz AVR state
+        :return: UC API remote state
+        """
+        if avr_state in REMOTE_STATE_MAPPING:
+            return REMOTE_STATE_MAPPING[avr_state]
+        return ucapi.remote.States.UNKNOWN
+
+    def filter_changed_attributes(self, update: dict[str, Any]) -> dict[str, Any]:
+        """
+        Filter the given attributes and return only the changed values.
+
+        :param update: dictionary with attributes.
+        :return: filtered entity attributes containing changed attributes only.
+        """
+        attributes = {}
+
+        if Attributes.STATE in update:
+            state = DenonRemote.state_from_avr(update[Attributes.STATE])
+            attributes = helpers.key_update_helper(Attributes.STATE, state, attributes, self.attributes)
+
+        return attributes
+
+    async def _handle_command(self, cmd_id: str, params: dict[str, Any] | None = None) -> StatusCodes:
+        command = ""
+        if params:
+            command = params.get("command", "")
+
+        if command == "":
+            command = f"remote.{cmd_id}"
+
+        match command:
+            case "remote.send":
+                return await self._denon_media_player.command(media_player.Commands.ON)
+            case "remote.off":
+                return await self._denon_media_player.command(media_player.Commands.OFF)
+            case "remote.toggle":
+                return await self._denon_media_player.command(media_player.Commands.TOGGLE)
+
+        if cmd_id == Commands.SEND_CMD:
+            return await self._denon_media_player.command(command)
+
+        return StatusCodes.NOT_IMPLEMENTED
```

Comment:
- Command `send_cmd_sequence` is missing
- Optional `repeat` parameter is not handled